xlators/mgmt/glusterd/src/glusterd.h

```diff
@@ -764,18 +764,14 @@ typedef ssize_t (*gd_serialize_t)(struct iovec outmsg, void *args);
     } while (0)
 
 #define RCU_READ_LOCK                                                          \
-    pthread_mutex_lock(&(THIS->ctx)->cleanup_lock);                            \
```

Comment:
Can these locks be removed safely ? I guess they were added for some reason...

Reply:
This code was introduced by this patch(https://review.gluster.org/#/c/glusterfs/+/21743/) to avoid a glusterd crash at the time of reboot and big_handler call at the same time.I believe the approach was not correct, will send a new patch to handle the cleanup during any actor function calling.

Reply:
Have you checked how difficult it would be to correctly terminate requests before calling xlator's `fini()` ?